src/MLJGLMInterface.jl

```diff
@@ -327,4 +327,301 @@ metadata_model(
+   probabilistic.
+- `predict_mean(mach, Xnew)`: instead return the mean of
+   each prediction above
+
+# Fitted parameters
+
+The fields of `fitted_params(mach)` are:
+
+- `features`: The names of the features used during model fitting
+- `coef`: The linear coefficients determined by the model
+- `intercept`: The intercept determined by the model
+
+# Report
+
+The fields of `report(mach)` are:
+
+- `deviance`: Measure of deviance of fitted model with respect to
+   a perfectly fitted model. For a linear model, this is the weighted
+   residual sum of squares
+- `dof_residual`: degrees of freedom for residuals, when meaningful
+- `stderror`: standard errors of the coefficients
+- `vcov`: estimated variance-covariance matrix of the coefficient estimates
+
+# Examples
+
+```
+using MLJ
+GLM = @load LinearRegressor pkg=GLM
+glm = GLM()
+
+X, y = make_regression(100, 2) # synthetic data
+mach = machine(glm, X, y) |> fit!
+
+Xnew, _ = make_regression(3, 2)
+yhat = predict(mach, Xnew) # new predictions
+yhat_point = predict_mean(mach, Xnew) # new predictions
+
+fitted_params(mach).features
+fitted_params(mach).coef # x1, x2, intercept
+fitted_params(mach).intercept
+
+report(mach)
+```
+
+See also
+
+[`LinearCountRegressor`](@ref), [`LinearBinaryClassifier`](@ref)
+"""
+LinearRegressor
+
+"""
+$(MMI.doc_header(LinearBinaryClassifier))
+
+`LinearBinaryClassifier` is a [generalized linear model](https://en.wikipedia.org/wiki/Generalized_linear_model#Variance_function), specialised
+to the case of a binary target variable, with a user-
+specified link function. Options exist to specify an intercept or offset feature.
+
+
+# Training data
+
+In MLJ or MLJBase, bind an instance `model` to data with
+    mach = machine(model, X, y)
+
+Where
+
+- `X`: any table of input features (eg, a `DataFrame`) whose columns
+  each have one of the following element scitypes: `Continuous`,
+  `Count`, or `<:OrderedFactor`; check column scitypes with `schema(X)`
+
+- `y`: is the target, which can be any `AbstractVector` whose element
+  scitype is `<:OrderedFactor(2)` or `<:Multiclass(2)`; check the scitype
+  with `scitype(y)`
+
+# Hyper-parameters
+
+- `fit_intercept=true`: Whether to calculate the intercept for this model.
+   If set to false, no intercept will be calculated (e.g. the data is expected
+   to be centered)
+- `link = GLM.LogitLink`: The function which links the linear prediction function
+   to the probability of a particular outcome or class.
+- `offsetcol=nothing`: Name of the column to be used as an offset, if any.
+   An offset is a variable which is known to have a coefficient of 1.
+
+# Operations
+
+- `predict(mach, Xnew)`: return predictions of the target given
+  features `Xnew` having the same scitype as `X` above. Predictions
+  are probabilistic.
+- `predict_mean(mach, Xnew)`: return predictions of the target given
+  features `Xnew`. Predictions represent probaility of being the
+  "positive" (second) class.
```

Comment:
```suggestion
- `predict_mean(mach, Xnew)`: return the mean values of the probabilistic predictions
   returned above.
- `predict_mean(mach, Xnew)`: instead return median values.
```

Reply:
Thanks! I assume I should add this for each model, since it uses the MLJ fallback?

Reply:
Hmm, do we want to just add `predict_mode` docs, since we are defining a `predict_mean` function?

Reply:
This is a classifier, ie, the target values are categorical. So the mean is not defined, only mode.

For probabilistic regressors (there aren't many) we list `predict_mean` and `predict_median` as operations
For probabilistic classifiers we list `predict_mode`.
For deterministic models we don't list anything but `predict`. 

regressor: if target_scitype AbstractVector{<:Continuous}
classifier: if target_scitype AbstractVector{<:Finite} or Multiclass or OrderedFactor

Reply:
Makes sense!!